src/app/utils/helpers.ts

```diff
@@ -43,3 +45,37 @@ export const isValidTxOasisHash = (hash: string): boolean => /^[0-9a-fA-F]{64}$/
 export const isValidTxEthHash = (hash: string): boolean => /^0x[0-9a-fA-F]{64}$/.test(hash)
 
 export const isValidTxHash = (hash: string) => isValidTxOasisHash(hash) || isValidTxEthHash(hash)
+
+/**
+ * EIP-55 Mixed-case checksum address encoding
+ *
+ * See https://eips.ethereum.org/EIPS/eip-55
+ */
+export const toChecksumAddress = (address: string): string => {
+  address = address.toLowerCase().replace('0x', '')
+  const hash = createKeccakHash('keccak256').update(address).digest('hex')
+  let ret = '0x'
+
+  for (let i = 0; i < address.length; i++) {
+    if (parseInt(hash[i], 16) >= 8) {
+      ret += address[i].toUpperCase()
+    } else {
+      ret += address[i]
+    }
+  }
+
+  return ret
+}
```

Comment:
Please copy from web wallet instead

Reply:
Yes I considered that, but the `toChecksumAddress()` function I found there has this comment:

```
/**
 * Returns a checksummed address.
 *
 * If an eip1191ChainId is provided, the chainId will be included in the checksum calculation. This
 * has the effect of checksummed addresses for one chain having invalid checksums for others.
 * For more details see [EIP-1191](https://eips.ethereum.org/EIPS/eip-1191).
 *
 * WARNING: Checksums with and without the chainId will differ and the EIP-1191 checksum is not
 * backwards compatible to the original widely adopted checksum format standard introduced in
 * [EIP-55](https://eips.ethereum.org/EIPS/eip-55), so this will break in existing applications.
 * Usage of this EIP is therefore discouraged unless you have a very targeted use case.
 */
```

The key sentence being:

> the EIP-1191 checksum is **not backwards compatible to the original widely adopted checksum format** standard introduced in [EIP-55](https://eips.ethereum.org/EIPS/eip-55), so this will break in existing applications.

This is why I didn't want to use it.

Or is this  misunderstanding, and the incompatibility is only only present when one is adding the optional `chainId` ?

Reply:
> Or is this misunderstanding, and the incompatibility is only present when one is adding the optional chainId ?

I tested it, and yes, the results actually match. 
OK, so we can go with the code in [ethereumjs](https://www.npmjs.com/package/@ethereumjs/util) (used by the wallet)

Reply:
> Please copy from web wallet instead

Done.